### ISSUE TITLE
Inappropriate format of router ip in case of ipv6 traffic

### ISSUE THREAD

**Issue by piorek94** [phaag/nfdump#252]
### Problem:

Issue was observed for `nfcapd` tool (not known if related with other collecting tools). 

In case of generating statistic for flow element `router ip` results depends on type of traffic (ipv4/ipv6) regardless of output format. The first record in the file for the given IP address of the router is of particular importance. If first record is related with ipv6 traffic then router ip is given as IPv4-Compatible IPv6 Address, which lexically represents an address other than the strict/original ipv4 address (logically is the same) - this inconsistence could lead to errors.

### Description:

Exporting router has strict ipv4 address `163.19.55.2`. 

#### First record as ipv6
```
$ nfdump -r nfcapd.202010081631_anon -o raw -c 1 -q

Flow Record: 
  Flags        =              0x89 FLOW, Sampled
  label        =            <none>
  export sysid =                 2
  size         =               124
  first        =        1602167446 [2020-10-08 16:30:46]
  last         =        1602167446 [2020-10-08 16:30:46]
  msec_first   =               272
  msec_last    =               272
  src addr     =  193d:50..ff:0:96
  dst addr     =  dcfe:f5..ff:fffa
  src port     =               443
  dst port     =             50840
  fwd status   =                 0
  tcp flags    =              0x10 ...A....
  proto        =                 6 TCP  
  (src)tos     =                 0
  (in)packets  =               500
  (in)bytes    =            750000
  input        =              1266
  output       =              1205
  src as       =                 0
  dst as       =                 0
  src mask     =                48 193d:50..0:379::/48
  dst mask     =                56 dcfe:f5..:8e00::/56
  dst tos      =                 0
  direction    =               255
  ip next hop  =  5317:19..:0:ff00
  src vlan     =                 0
  dst vlan     =                 0
  ip router    =       163.19.55.2
  received at  =     1602167460049 [2020-10-08 16:31:00.049]
```

Statistic represents router ip address as IPv4-Compatible IPv6 Address:
```
$ nfdump -r nfcapd.202010081631_anon -s router -q
2020-10-02 16:49:59.808 517318.912 any       ::163.19.55.2     4461(93.2)    6.9 M(92.9)    7.2 G(94.9)       13   110806  1040
2020-10-08 16:30:06.592   112.384 any          50.73.15.8      328( 6.8)   530000( 7.1)  386.6 M( 5.1)     4715   27.5 M   729

$ nfdump -r nfcapd.202010081631_anon -s router -q -o pipe
10|1601650199|808|1602167518|720|0|0|0|37163939|0|4461|6885500|7165260500|13|110806|1040
2|1602167406|592|1602167518|976|0|0|0|0|843648776|328|530000|386613000|4715|27520857|729
```

#### First record as ipv4
```
$ nfdump -r nfcapd.202010081632_anon -o raw -c 1 -q

Flow Record: 
  Flags        =              0x80 FLOW, Sampled
  label        =            <none>
  export sysid =                 4
  size         =                92
  first        =        1602167508 [2020-10-08 16:31:48]
  last         =        1602167508 [2020-10-08 16:31:48]
  msec_first   =               992
  msec_last    =               992
  src addr     =     172.78.194.45
  dst addr     =      163.19.121.2
  src port     =               443
  dst port     =             52516
  fwd status   =                 0
  tcp flags    =              0x10 ...A....
  proto        =                 6 TCP  
  (src)tos     =                 0
  (in)packets  =               500
  (in)bytes    =             20000
  input        =              1266
  output       =              1082
  src as       =                 0
  dst as       =                 0
  src mask     =                24 172.78.194.0/24
  dst mask     =                24 163.19.121.0/24
  dst tos      =                 0
  direction    =               255
  ip next hop  =        212.36.5.3
  bgp next hop =        50.73.15.4
  src vlan     =                 0
  dst vlan     =                 0
  ip router    =       163.19.55.2
  received at  =     1602167520083 [2020-10-08 16:32:00.083]
```

Statistic represents router ip address as strict ipv4 address:
```
$ nfdump -r nfcapd.202010081632_anon -s router -q
2020-10-02 16:49:59.808 517379.584 any         163.19.55.2     5569(94.0)    6.5 M(92.4)    6.5 G(94.5)       12   100166  1002
2020-10-08 16:31:00.608   118.528 any          50.73.15.8      358( 6.0)   533000( 7.6)  373.8 M( 5.5)     4496   25.2 M   701

$ nfdump -r nfcapd.202010081632_anon -s router -q -o pipe
2|1601650199|808|1602167579|392|0|0|0|0|2735945474|5569|6462500|6477983500|12|100166|1002
2|1602167460|608|1602167579|136|0|0|0|0|843648776|358|533000|373788000|4496|25228671|701
```

### Notes
Aggregation deals with that issue but there is not option to output appropriate data in pipe format (which is mainly used)
```
$ nfdump -r nfcapd.202010081631_anon -A router -q
2020-10-02 16:49:59.808 517318.912       163.19.55.2     6.9 M    7.2 G   110806   1040  4461
2020-10-08 16:30:06.592   112.384        50.73.15.8    530000  386.6 M   27.5 M    729   328
$ nfdump -r nfcapd.202010081631_anon -A router -q -o pipe
10|1601650199|808|1602167518|720|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|6885500|7165260500
2|1602167406|592|1602167518|976|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|530000|386613000

$ nfdump -r nfcapd.202010081632_anon -A router -q
2020-10-02 16:49:59.808 517379.584       163.19.55.2     6.5 M    6.5 G   100166   1002  5569
2020-10-08 16:31:00.608   118.528        50.73.15.8    533000  373.8 M   25.2 M    701   358
$ nfdump -r nfcapd.202010081632_anon -A router -q -o pipe
2|1601650199|808|1602167579|392|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|6462500|6477983500
2|1602167460|608|1602167579|136|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|533000|373788000
```

### Expected behavior

If exporting router has ipv4 address it should be represented as strict ipv4 address (not as IPv4-Compatible IPv6 Address), for all output formats while generating statistic (`-s router/...` option) regardless of the type of traffic (ipv4/ipv6).

**Comment by phaag** [comment]
Thanks for the bug report.

**Comment by phaag** [comment]
Fixed in commit 353d6ff
